Sources/System/FileHelpers.swift

```diff
@@ -122,4 +122,18 @@ extension FileDescriptor {
       return .success(buffer.count)
     }
   }
+
+  /// Return the current size of the file.
+  ///
+  /// - Returns: The current size of the file, in bytes.
+  @_alwaysEmitIntoClient
+  @discardableResult
+  public func fileSize(
+    retryOnInterrupt: Bool = true
+  ) throws -> Int64 {
+    let current = try seek(offset: 0, from: .current)
+    let size = try seek(offset: 0, from: .end)
+    try seek(offset: current, from: .start)
```

Comment:
This is temporarily mutating the file position during what looks like a read-only operation; I feel the potential pitfalls of this (alongside the fact that many file descriptors do not have a file size) probably makes this entry point less suitable for this package.

Reply:
That's reasonable. I added the following fixup to address this:

* d90bf85 `fixup: Move FileDescriptor.fileSize() to an internal extension in SystemTests`